### PR TITLE
RD-8833-spurious-awsecsdetector-log-in-python-distro

### PR DESCRIPTION
**Pull request by shanishiri**

### CODE REVIEW

File: src/lumigo_opentelemetry/resources/detectors.py [hunk 1]
```diff
@@ -102,13 +103,18 @@ def detect(self) -> "Resource":
...
-            AwsEcsResourceDetector(),
-            LumigoAwsEcsResourceDetector(),
-        ],
+        detectors=_get_detector_list(),
         initial_resource=Resource.create(attributes=attributes),
     )
+
+
+def _get_detector_list():
+    logging.getLogger("opentelemetry.sdk.extension.aws.resource.ecs").setLevel(logging.CRITICAL)
```
Comment (mmanciop) [inline src/lumigo_opentelemetry/resources/detectors.py:112]: It feels to me like this code should be in the beginning of the distro init, next to the other code we have for loggers

File: src/lumigo_opentelemetry/resources/detectors.py [hunk 2]
```diff
@@ -102,13 +103,18 @@ def detect(self) -> "Resource":
...
-            LumigoDistroDetector(),
-            AwsEcsResourceDetector(),
-            LumigoAwsEcsResourceDetector(),
-        ],
+        detectors=_get_detector_list(),
         initial_resource=Resource.create(attributes=attributes),
     )
+
+
+def _get_detector_list():
```
Comment (nirLumigo) [inline src/lumigo_opentelemetry/resources/detectors.py:111]: Can you add return type to the function?

**Review by nirLumigo — APPROVED** [review]

File: src/lumigo_opentelemetry/resources/detectors.py (hunk 1 above)
Comment (shanishiri) [reply src/lumigo_opentelemetry/resources/detectors.py]: I think here it's more specific for the detector and not like in the init() where it's more general. Also thinking about the future - if we'll need to suppress other detectors logs then the right place to do it is here

File: src/lumigo_opentelemetry/resources/detectors.py [hunk 3]
```diff
@@ -102,13 +104,20 @@ def detect(self) -> "Resource":
...
-            AwsEcsResourceDetector(),
-            LumigoAwsEcsResourceDetector(),
-        ],
+        detectors=_get_detector_list(),
         initial_resource=Resource.create(attributes=attributes),
     )
+
+
+def _get_detector_list() -> List[ResourceDetector]:
+    logging.getLogger("opentelemetry.sdk.extension.aws.resource.ecs").setLevel(
```
Comment (GuyMoses) [inline src/lumigo_opentelemetry/resources/detectors.py:113]: check if `getLogger` returns optional - so that `.setLevel.` won't raise

**Review by GuyMoses — APPROVED** [review]

File: src/lumigo_opentelemetry/resources/detectors.py (hunk 3 above)
Comment (shanishiri) [reply src/lumigo_opentelemetry/resources/detectors.py:113]: if the logger does not exist the getLogger will create it

File: src/lumigo_opentelemetry/resources/detectors.py (hunk 1 above)
Comment (mmanciop) [reply src/lumigo_opentelemetry/resources/detectors.py]: Yes but the question is going to be “why don’t I see XYZ logs”, and people will naturally first look at https://github.com/lumigo-io/opentelemetry-python-distro/blob/925285d811ef6a3fe5def4b7b8503dae7241378d/src/lumigo_opentelemetry/__init__.py#L10. That is where this code should live with a comment about it.

**Pull request merged**